server/src/main/java/org/elasticsearch/cluster/metadata/Template.java

```diff
@@ -305,6 +355,7 @@ public static class Builder {
         private CompressedXContent mappings = null;
         private Map<String, AliasMetadata> aliases = null;
         private DataStreamLifecycle lifecycle = null;
+        private ExplicitlyNullable<DataStreamOptions.Template> dataStreamOptions = null;
```

Comment:
Example: This would become something like `ExplicitlyNullable<>.unspecified()` instead of `null`